### PR TITLE
Redesign state badge widget on tiles

### LINKED ISSUE #303: Redesign state badge widget on tiles
Here's a mockup of a redesigned state badge widget.

* the design in general is inspired by the state badges in the footer of the state badges in object lists.
* number of total node will be moved next to the badges

Artboard

### PR DESCRIPTION
**Pull request by raviks789**
resolves #303 

requires https://github.com/Icinga/ipl-web/pull/63

### CODE REVIEW

File: public/css/state-badges.less [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      border-top-right-radius: 0;
+    }
+
+    li > ul > li:last-child:not(:first-child) .state-badge {
+      border-bottom-left-radius: 0;
+      border-top-left-radius: 0;
+    }
+
+    > li:not(:last-child) {
+      margin-right: .25em;
```
Comment (flourish86) [inline public/css/state-badges.less:24]: To save space I'd go for `1px` for `margin-right`.

**Review by flourish86 — APPROVED** [review]
Looks perfect to me. Thanks! 👍

**Review by nilmerg — CHANGES_REQUESTED** [review]
I've ensured compatibility with v2.10 now and cleaned up a bit. Note that https://github.com/Icinga/ipl-web/pull/63 is required for this to work now.

Though, I believe some changes in the UI are not thought through enough:
* The total children count is placed at a weird location. The count belongs to the *direct* children. It's not the total of all states that are shown on the right. Best to be seen in the lower left tile. I think this confuses users an we'll get certainly reports that this number is too low. I think we should place the total count where it was previously.
* The *OK* states we're not shown previously. Now if there's everything green, like in the tiles below, they look somewhat off to me. I'd propose to not show them like before.

Screenshot from 2021-12-17 13-44-31

**Comment by flourish86** [comment]
As discussed in our meeting, we should change the badges counts to the following:

* We will only show problems, so everything, but ok/up
* We will refer to nodes (Host/Service/Process) for the states, so there will be max. one badge for every state
* As a consequence the sum of the state badges items will be max. the number of the total amount.
* To clarify this we label the total number of node “[n] *Nodes*”

This is how this would then for example look in the end.

Artboard Copy 6

File: library/Businessprocess/Renderer/Renderer.php [hunk 2]
```diff
@@ -135,34 +136,42 @@ public function countChildNodes()
...
+                    'Child',
+                    'Children',
+                    $totalChildren,
+                    'businessprocess.nodes'
+                ))
+            ],
+            [
+                $totalChildren,
+                ' ',
+                mtp('node', 'Node', 'Nodes', $totalChildren)
```
Comment (nilmerg) [inline library/Businessprocess/Renderer/Renderer.php:160]: The first argument is the domain, which should be the module's name. (i.e. `businessprocess`)

Also, please use the terms `Child` and `Children` instead. Best to be seen in the title above, which is redundant then btw. Just use the very same instructions.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Businessprocess/Renderer/Renderer.php (hunk 2 above)
Comment (nilmerg) [reply library/Businessprocess/Renderer/Renderer.php]: Why haven't you removed the title? Isn't it redundant?

The amount of children is not part of the translated message. It must be. Translators should be able to move it: `mtp('foo', 'just one', 'many of them (%u)', $count)`

Comment (flourish86) [reply library/Businessprocess/Renderer/Renderer.php]: ✅

**Review by nilmerg — APPROVED** [review]

**Pull request merged**